### PR TITLE
feat: treesitter checkhealth

### PR DESCRIPTION
**Pull request by teto**
nvim-treesitter has a neat checkhealth already but it would be neat to have a basic one in core (mustered the motivation with the recent ABI breaking changes).

### CODE REVIEW

**Comment by runiq** [comment]
+1, this would have been great during the recent 0.19 kerfuffle :)

**Comment by theHamsta** [comment]
Regarding the ABI, nvim gets this information from API at the time when it was built. Some distros started serving shared libs of tree-sitter which silently change nvim's runtime without this being visible with the current mechanism. Although linking with an incompatible shared lib is doomed to break silently from the beginning.

**Comment by vigoux** [comment]
I'll try to take over the work on this pr, to have that ready for 0.5 maybe.
@teto do you think it's worth moving to the 0.5 milestone?

**Comment by teto** [comment]
nice. I hope it can make it for 0.5 but as anything treesitter, I dont think it should block it

**Comment by teto** [comment]
NB: nvim-treesitter looks for a list of predefined grammars, I prefer the approach in this PR where a regex tries to match any grammar. It's more generic.

**Comment by vigoux** [comment]
Fairly happy about how that turned out, I think it is good to merge after a little review, tell me what you think @teto

**Comment by teto** [comment]
works for me looks good

File: runtime/lua/vim/treesitter/health.lua [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+function M.check_health()
+  local report_info = vim.fn['health#report_info']
+  local report_ok = vim.fn['health#report_ok']
+  local report_error = vim.fn['health#report_error']
+  local parsers = M.list_parsers()
+
+  report_info(string.format("Runtime ABI version : %d", ts.language_version))
+
+
```
Comment (mhinz) [inline runtime/lua/vim/treesitter/health.lua:16]: Superfluous blank line. :)

**Review by mhinz — APPROVED** [review]

**Pull request merged**